internal/db/queryBuilder.go

```diff
@@ -105,7 +106,12 @@ func (qb *queryBuilder) updatedObjectsCTE() string {
 		ignoresPredicate = "AND o.path NOT LIKE ALL(__ignores__::text[])"
 	}
 
-	return fmt.Sprintf(template, isCachedSelector, cacheJoin, pathPredicate, ignoresPredicate)
+	subpathsPredicate := ""
+	if len(qb.objectQuery.Subpaths) > 0 {
+		subpathsPredicate = "AND o.path LIKE ALL(__subpaths__::text[])"
+	}
```

Comment:
I guess this option isn't mutually exclusive with `ignores` right -- you could use both at once? And we'll just let PG decide which conditions to use the index for?

Reply:
You can use both at once, ignoring a subset of the files under the subpaths. I initially wanted to use it like (haskell's lens)[https://github.com/ekmett/lens/wiki/Overview] functionality where the subpaths focus you on a specific set of the project and let you manipulate just that. With that in mind I thought keeping the ignores made sense.